# 3.03 Math Expressions: Completed Homework

**Adhvay Iyer · RazorCrest00 · September 22, 2026**

Solutions for [3.03 Math Expressions](https://pages.opencodingsociety.com/csp/python/math-expressions/lesson): both Popcorn hacks and the recovery-center report. The live lesson describes costs in Popcorn 2 but supplies a capacity starter, so both versions are completed below.

Every Python cell runs independently. Click **Run** to reproduce the saved notebook output; edit the starting values and rerun to explore a different case. Initial browser execution downloads Python once.

[Download notebook](https://raw.githubusercontent.com/RazorCrest00/portfolio/main/_notebooks/homework/2026-09-22-math-expressions-hw.ipynb) · [GitHub source](https://github.com/RazorCrest00/portfolio/blob/main/_notebooks/homework/2026-09-22-math-expressions-hw.ipynb)

The numbers are classroom examples, not a statement of the center's actual prices or capacity.

## Popcorn Hack 1: Basic arithmetic

**Expected results:** total `135`, difference `90`, product `250`, quotient `10.0`. Each result is computed from the supplied variables, so changing an input changes the calculation.

In [1]:
# CODE_RUNNER: Popcorn 1 - Basic arithmetic
value_a = 100
value_b = 10
value_c = 25

addition = value_a + value_b + value_c
subtraction = value_a - value_b
multiplication = value_b * value_c
division = value_a / value_b

print("Total:", addition)
print("Difference:", subtraction)
print("Product:", multiplication)
print("Division:", division)

Total: 135
Difference: 90
Product: 250
Division: 10.0


**Why:** Addition combines all three values; subtraction keeps the requested A-minus-B order. Multiplication uses B and C. Python `/` produces a float, which explains `10.0` even though the division is exact. The divisor must not be zero.

## Popcorn Hack 2A: Session costs

This completes the written cost task. **Expected:** `$105.00` per week and `$420.00` for the four-week model. Multiplying dollars/session by sessions/week produces dollars/week; multiplying again by weeks/month produces dollars/month.

In [2]:
# CODE_RUNNER: Popcorn 2A - Weekly and monthly session costs
cost_per_session = 35
sessions_per_week = 3
weeks_per_month = 4

weekly_cost = cost_per_session * sessions_per_week
monthly_cost = weekly_cost * weeks_per_month

print(f"Weekly Cost: ${weekly_cost:.2f}")
print(f"Monthly Cost: ${monthly_cost:.2f}")

Weekly Cost: $105.00
Monthly Cost: $420.00


## Popcorn Hack 2B: Booking capacity

This completes the supplied capacity starter. **Expected:** `12` occupied spots, `88` remaining spots, and `44.0` per room. Multiplication groups the bookings before subtraction. The division is an even split for this sample, not an algorithm for assigning individual people when a remainder exists.

In [3]:
# CODE_RUNNER: Popcorn 2B - Recovery-center booking capacity
total_monthly_capacity = 100
spots_requested_per_group = 4
sessions_booked = 3
active_rooms = 2

total_spots_taken = spots_requested_per_group * sessions_booked
remaining_capacity = total_monthly_capacity - total_spots_taken
users_per_room = remaining_capacity / active_rooms

print("Total Spots Taken:", total_spots_taken)
print("Remaining Capacity:", remaining_capacity)
print("Users per Room:", users_per_room)

Total Spots Taken: 12
Remaining Capacity: 88
Users per Room: 44.0


## Homework: Recovery Center Cost Report

| Result | Calculation | Expected value |
| --- | --- | --- |
| Weekly cost | 35 × 3 | $105.00 |
| Monthly cost | 105 × 4 | $420.00 |
| First month including enrollment | 420 + 75 | $495.00 |
| Average daily session cost | 420 ÷ 30 | $14.00 |

The enrollment fee is added **once**, after the monthly session cost. The assignment defines the daily average using session costs only, so the fee is excluded from that average. Four weeks and 30 days are the provided model assumptions.

In [4]:
# CODE_RUNNER: Homework - Complete cost report
cost_per_session = 35
sessions_per_week = 3
weeks_per_month = 4
enrollment_fee = 75
days_per_month = 30

weekly_cost = cost_per_session * sessions_per_week
monthly_cost = weekly_cost * weeks_per_month
total_cost_with_fee = monthly_cost + enrollment_fee
average_daily_cost = monthly_cost / days_per_month

print(f"Weekly Cost: ${weekly_cost:.2f}")
print(f"Monthly Cost: ${monthly_cost:.2f}")
print(f"Total Cost with Fee: ${total_cost_with_fee:.2f}")
print(f"Average Daily Cost: ${average_daily_cost:.2f}")

Weekly Cost: $105.00
Monthly Cost: $420.00
Total Cost with Fee: $495.00
Average Daily Cost: $14.00


**Reasoning:** The later expressions reuse earlier results instead of repeating numbers. `:.2f` formats the output as currency without changing the stored value. A different fee affects only the first-month total; a different session count affects all session-cost calculations. Assignments store results, so changing an input requires rerunning the calculations.

For example, including the fee in the daily calculation would give `$16.50`, which answers a different question. Multiplying the enrollment fee by four would also be incorrect because it is not a weekly charge.

## Extension: Precedence, powers, and remainders

The required solutions use `+`, `-`, `*`, and `/`. This short extension covers the lesson's remaining operators and shows why grouping matters. `/` performs ordinary division, `//` rounds the quotient down, and `%` returns the remainder. For these positive operands, `17 = 5 × 3 + 2`.

In [5]:
# CODE_RUNNER: Extension - Check operator behavior
print("Multiplication first:", 2 + 3 * 4)
print("Parentheses first:", (2 + 3) * 4)
print("Power:", 2 ** 3)
print("Full groups:", 17 // 3)
print("Remainder:", 17 % 3)
print("Negative floor quotient:", -17 // 3)
print("Negative remainder:", -17 % 3)

assert 2 + 3 * 4 == 14
assert (2 + 3) * 4 == 20
assert 2 ** 3 == 8
assert (17 // 3) * 3 + 17 % 3 == 17
assert (-17 // 3) * 3 + (-17 % 3) == -17
print("Operator checks: PASS")

Multiplication first: 14
Parentheses first: 20
Power: 8
Full groups: 5
Remainder: 2
Negative floor quotient: -6
Negative remainder: 1
Operator checks: PASS


**Expected:** `14`, `20`, `8`, `5`, `2`, `-6`, `1`, followed by a passing check. The negative example distinguishes floor division from truncating toward zero.

**RPN note:** A stack is **last in, first out**. For `5 3 - 2 *`, pop the right operand (`3`) before the left (`5`), compute `5 - 3`, then multiply the result by `2` to obtain `4`. Operand order matters for subtraction and division. See [Python's stack explanation](https://docs.python.org/3/tutorial/datastructures.html#using-lists-as-stacks).

## Verification: Change the inputs

These results came from rerunning the **actual homework cell** with only its starting inputs changed. Each case checked all four calculated values and the four labeled output lines.

| Test | Weekly | Monthly | With fee | Daily | Result |
| --- | --- | --- | --- | --- | --- |
| Required sample | $105.00 | $420.00 | $495.00 | $14.00 | PASS |
| No sessions | $0.00 | $0.00 | $75.00 | $0.00 | PASS |
| No enrollment fee | $105.00 | $420.00 | $420.00 | $14.00 | PASS |
| One session per week | $35.00 | $140.00 | $215.00 | $4.67 | PASS |
| Five-week model | $105.00 | $525.00 | $600.00 | $17.50 | PASS |
| Fractional session price | $106.50 | $426.00 | $501.00 | $14.20 | PASS |

The zero-session case keeps the fee because the formula always adds it; this is a test of the assigned model, not a real billing policy. The one-session case stores the full `140 / 30` result but displays `$4.67`. To reproduce a row, replace the named input and rerun the homework cell. Restore the supplied inputs before submitting.

All five cells were also executed separately with fresh variables and their output saved in this notebook. No input prompts or imports are required. These examples assume valid numeric inputs, nonnegative counts, and nonzero divisors.

## Rubric evidence and self-assessment

| Criterion | Available | Evidence |
| --- | --- | --- |
| Popcorn 1 | 0.2 | Four correct expressions and labeled results. |
| Popcorn 2 | 0.2 | Both the written cost exercise and the capacity starter completed. |
| Homework weekly/monthly | 0.2 | Variable-based calculations produce 105 and 420. |
| Homework fee/daily average | 0.2 | A one-time addition produces 495; session-only daily average is 14. |
| Runs and labeled results | 0.2 | Every cell executes independently and labels each result; saved outputs and six homework test cases provide evidence. |
| **Coverage** | **1.0** | All five rubric criteria are addressed. |

**Why this supports full credit:** Every required result is calculated from variables and can be reproduced. The explanations distinguish recurring costs from the one-time fee, clarify units and formatting, and demonstrate the additional arithmetic operators. Changed-input checks show that the formulas are not hardcoded to the sample. This maps the work to the rubric; the final score remains the grader's decision.

**References and attribution:** Exercise requirements and starting data come from the linked [OCS lesson](https://pages.opencodingsociety.com/csp/python/math-expressions/lesson), checked September 22, 2026. Operator behavior follows [Python's official numeric tutorial](https://docs.python.org/3/tutorial/introduction.html#numbers). Solutions, explanations, and verification were prepared with AI assistance; no peer review is claimed.

**Submission:** Use this page's published URL in the lesson's [Submit Homework form](https://forms.gle/mi5cLycuyjdaZo5t7). The notebook download is available above if a file is requested.

<script>
// This submission contains Python solutions only.
document.addEventListener('DOMContentLoaded', () => {
  document.querySelectorAll('.code-runner-container .languageSelect').forEach(select => {
    select.replaceChildren(new Option('Python', 'python'));
  });
});
</script>